docs/design/design_vuln_scan_job_in_same_namespace_of_workload.md

```diff
@@ -0,0 +1,81 @@
+# Run Vulnerability Scan Job in Same namespace of workload
+
+## Overview 
```

Comment:
Running scan jobs in the same namespace as workload is an implementation detail or a prerequisite to scan particular workloads that use images from private managed registries.

What we should clarify is why we even consider running scan jobs in the same namespace. In particular, we should explain how ECR injects credentials as extra environments variables passed to containers that run with authorized service account. IIRC in EKS integrated with ECR there is this mutating webhook that does the trick. So we want to run scan jobs in the same namespace because we want to use workload's service account instead of the `starboard-operator` service account. By doing that the mutating webhooks will add some extra variables to scan containers (created by Starboard) that are eventually passed to Trivy. Finally, Trivy can recognise injected AWS environment variables and use to pull down private image from managed registry. Thus, imagePullSecrets are not required.

Another thing I'd like us to clarify is Trivy FS scanner and why we consider it along with same namespace config / design proposal? I remember we discussed it that if we pass image pull secrets to Trivy FS scanner it somehow falls back to the current behaviour, where Trivy pull image from the remote registry and is not scanning image cached on K8s node.

Reply:
thanks @danielpacak , I have added some details to explain the motivation for this proposal. Please review it again.